pkg/apis/garden/validation/validation.go

```diff
@@ -429,6 +430,77 @@ func validateAzureDomainCount(domainCount []garden.AzureDomainCount, fldPath *fi
 	return allErrs
 }
 
+////////////////////////////////////////////////////
+//                    PROJECTS                    //
+////////////////////////////////////////////////////
+
+// ValidateProject validates a Project object.
+func ValidateProject(project *garden.Project) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, apivalidation.ValidateObjectMeta(&project.ObjectMeta, false, ValidateName, field.NewPath("metadata"))...)
+	allErrs = append(allErrs, validateNameConsecutiveHyphens(project.Name, field.NewPath("metadata", "name"))...)
+	allErrs = append(allErrs, ValidateProjectSpec(&project.Spec, field.NewPath("spec"))...)
+
+	return allErrs
+}
+
+// ValidateProjectUpdate validates a Project object before an update.
+func ValidateProjectUpdate(newProject, oldProject *garden.Project) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, apivalidation.ValidateObjectMetaUpdate(&newProject.ObjectMeta, &oldProject.ObjectMeta, field.NewPath("metadata"))...)
+	allErrs = append(allErrs, ValidateProject(newProject)...)
+
+	return allErrs
+}
+
+// ValidateProjectSpec validates the specification of a Project object.
+func ValidateProjectSpec(projectSpec *garden.ProjectSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	allErrs = append(allErrs, ValidateSubject(projectSpec.Owner, fldPath.Child("owner"))...)
+	if description := projectSpec.Description; description != nil && len(*description) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("description"), "must provide a description when key is present"))
+	}
+	if purpose := projectSpec.Description; purpose != nil && len(*purpose) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("purpose"), "must provide a purpose when key is present"))
+	}
+
+	return allErrs
+}
+
+// ValidateSubject validates the subject representing the owner.
+func ValidateSubject(subject rbacv1.Subject, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(subject.Name) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("name"), ""))
+	}
+
+	switch subject.Kind {
+	case rbacv1.ServiceAccountKind:
+		allErrs = append(allErrs, field.Forbidden(fldPath.Child("kind"), fmt.Sprintf("kind %s is not allowed for the owner", subject.Kind)))
```

Comment:
This restriction breaks the possibility for full automation. I might have a fully automated CI/CD pipeline where only a `ServiceAccount` can do those operations.

Reply:
Agreed, removed this validation step.